.github/workflows/end_to_end_tests.yml

```diff
@@ -43,6 +43,38 @@ jobs:
           do_not_skip: "${{ steps.get_events_to_not_skip.outputs.events_to_not_skip }}"
           github_token: ${{ github.token }}
 
+  coverage:
```

Comment:
It probably makes sense to eventually refactor those steps which are duplicated across multiple workflows in a separate standalone re-usable GHA action.

Reply:
Agreed.